auto-value-javabean/src/main/java/org/graylog/autovalue/WithBeanGetterExtension.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.autovalue;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.extension.AutoValueExtension;
+import com.google.common.base.CaseFormat;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import java.util.Map;
+
+/**
+ * Creates Java Bean getter methods for each property.
+ */
+public class WithBeanGetterExtension extends AutoValueExtension {
+    private static final ClassName ANNOTATION_JSON_PROPERTY = ClassName.get(JsonProperty.class);
```

Comment:
I have been using the `ClassName.get("com.fasterxml.jackson.annotation", "JsonIgnore");` version in the original code so we don't have to depend on jackson. If we want to extract this extension into a separate project later - which I think is a good idea because this might be useful to others as well - we should probably use that code again.